src/chapters/orm-relationships/many-to-many.rst

```diff
@@ -0,0 +1,332 @@
+
+This form has two inputs. The first---with ``th:field="${eventTag.event}"``---is hidden, since it should not be modified by the user. We use it to keep track of the specific event that we are about to add a tag to.
+
+The second field, the ``select`` element, is bound to the ``eventTag.tag`` field. The dropdown contains each of the available tags in our application. 
+
+When this form is submitted, it will have all of the information necessary to create an ``EventTagDTO`` object using model binding.
+
+Processing the Form
++++++++++++++++++++
+
+As with ``displayAddTagForm``, we will break down the form's ``POST`` handler in ``EventController`` in detail.
+
+.. sourcecode:: java
+   :lineno-start: 124
+
+   @PostMapping("add-tag")
+   public String processAddTagForm(@ModelAttribute @Valid EventTagDTO eventTag,
+                                 Errors errors,
+                                 Model model){
+
+      if (!errors.hasErrors()) {
+         Event event = eventTag.getEvent();
+         Tag tag = eventTag.getTag();
+         if (!event.getTags().contains(tag)){
+               event.addTag(tag);
+               eventRepository.save(event);
+         }
+         return "redirect:detail?eventId=" + event.getId();
+      }
+
+      return "redirect:add-tag";
+   }
+
+Using model binding, our method takes a valid parameter of type ``EventTagDTO``. Since we referenced the ``event`` and ``tag`` fields of our DTO when rendering the form (in the template, using ``th:field``), our submitted form should contain all of the data necessary to create an ``EventTagDTO`` instance. This instance will be valid if both ``event`` and ``tag`` are non-null. 
+
+Lines 130-131 retrieve the values of these fields from the DTO. Then, as long as the given event doesn't already have the given tag, we add the tag to it's collection in lines 132-134. Finally, we save the ``event`` to ``eventRepository``, which results in the relationship being stored in the database.
+
+Exactly *how* this relationship is stored utilizes a new type of SQL table.
+
+Join Tables
+^^^^^^^^^^^
+
+Think about how relationships are established at the database level. One-to-one and one-to-many relationships are facilitated by the use of a foreign key column on one side of the relationship. Our ``event`` table has two foreign key columns: ``event_category_id`` and ``event_details_id``. 
+
+For a given row in ``event``, the column ``event_category_id`` contains the primary key of the row in ``event_category`` that the ``event`` row is related to, an similarly for ``event_details_id``. 
+
+The only difference is the *number* of different ``event`` rows that may have the same value of ``event_category_id`` and ``event_details_id``. The ``event``/``event_category`` relationship is many-to-one, so *many* event rows may have the same ``event_category_id`` value. The ``event``/``event_details`` relationship is one-to-one, so *only one* event row may have a given value in ``event_details_id``. 
```

Comment:
I think replacing the use of "given" in the last sentence with "certain" might make this point clearer that it has to do with a specific value, not just having a provided value